core/src/main/scala/com/yahoo/maha/core/query/QueryPipeline.scala

```diff
@@ -617,17 +625,21 @@ object DefaultQueryPipelineFactory extends Logging {
 
 class DefaultQueryPipelineFactory(implicit val queryGeneratorRegistry: QueryGeneratorRegistry, druidMultiQueryEngineList: Seq[Engine] = DefaultQueryPipelineFactory.druidMultiQueryEngineList) extends QueryPipelineFactory with Logging {
 
-  private[this] def getDimOnlyQuery(bestDimCandidates: SortedSet[DimensionBundle], requestModel: RequestModel): Query = {
+  private[this] def getDimOnlyQuery(bestDimCandidates: SortedSet[DimensionBundle], requestModel: RequestModel, queryGenVersion: Version): Query = {
     val dimOnlyContextBuilder = QueryContext
       .newQueryContext(DimOnlyQuery, requestModel)
       .addDimTable(bestDimCandidates)
     require(bestDimCandidates.nonEmpty, "Cannot generate dim only query with no best dim candidates!")
     require(queryGeneratorRegistry.isEngineRegistered(bestDimCandidates.head.dim.engine)
       , s"Failed to find query generator for engine : ${bestDimCandidates.head.dim.engine}")
-    queryGeneratorRegistry.getGenerator(bestDimCandidates.head.dim.engine).get.generate(dimOnlyContextBuilder.build())
+    queryGeneratorRegistry.getGenerator(bestDimCandidates.head.dim.engine, Some(queryGenVersion)).get.generate(dimOnlyContextBuilder.build())
```

Comment:
Wrap it in Option instead of Some, in case someone passes in null... Option(queryGenVersion)